### DISCUSSION TITLE
Returning a value in host functions

### DISCUSSION THREAD

**Discussion by omkar-mohanty** [WasmEdge/WasmEdge#2133] [Q&A]
I am writing OpenCV plugin for WasmEdge. I have a question regarding how to return values in a plugin function. For example `imread`  returns an object of type `Mat` so how am I supposed to return that. From the other plugins that I have seen(crypto and httpsreq) the `body` member function for a plugin  just returns an `int`.

**Comment by hydai** [comment]
You can get some ideas from the WASI-NN plugin: https://github.com/WasmEdge/WasmEdge/blob/master/plugins/wasi_nn/wasinnfunc.h#L36-L42

All of the non-primitive types should be encoded as a pointer and the corresponding length.
